workflow-core/src/jsMain/kotlin/com.squareup.workflow1/WorkflowIdentifierTypeNamer.kt

```diff
@@ -0,0 +1,23 @@
+package com.squareup.workflow1
+
+import kotlin.reflect.KClass
+
+public actual object WorkflowIdentifierTypeNamer {
+  // Stores mappings between KClass instances and their assigned names.
+  val mappings = mutableMapOf<KClass<*>,String>()
+
+  public actual fun uniqueName(kClass: KClass<*>): String {
+    // Note: `kClass.qualifiedName` cannot be used here like other platforms as it's not supported
+    // for JS. Therefore, we construct a unique name of each static KClass based on its simple name
+    // and an index of when it was encountered.
+
+    val mapping = mappings[kClass]
+    if (mapping != null) {
+      return mapping
+    }
+
+    val identifier = "${kClass.simpleName ?: kClass.hashCode()}(${mappings.size})"
```

Comment:
Note: This name doesn't differentiate against a genericized Workflow (although IIRC this same problem exists with `kClass.qualifiedName` in the native implementations too). Should I have put a comment here to mention this limitation?

Reply:
Definitely put the comment about the limitations. Where were you seeing this problem for the kotlin native code? Do you have a link I can read more about this?
Was this a suggested solution for this with kotlin-js? Or something you invented?

Reply:
If I run the following against the JVM, it doesn't include the generic type:

```
data class GenericType<T>(
  val t: T
)

val someType = GenericType<String>(t = "abc")
println(someType::class.qualifiedName) // prints "com.squareup.sandbox.GenericType"
```

As for this implementation, I invented this solution. This seems to work in practice since it appears Kotlin/JS uses a static class (probably stored on its prototype) to represent the KClass. Open to suggestions though!

Reply:
Ah I C what you mean now.  Usuallly that is when the identifier would be overridden I believe.